data/libs/Calibration.lua

```diff
@@ -0,0 +1,118 @@
+	local a = Calibration.PriceExponential
+	local constant = Calibration.StockPriceOne
+	local coeff = (v2 * constant / cargo)^((1-a) / a)
+	local profitTo = p * v * coeff * constant
+	local profitFrom = p * v2 * coeff * constant / 2
+	return (profitTo + profitFrom) / 2
+end
+
+--Define the ratio of fighting rewards to minimum trading profit in linear way. Used in Assassination and Combat modules.
+Calibration.Fighting = function (missiles)
+	local ratio = Rab^(Nships - 1)
+	return math.max(1, (ratio - 1) / (maxMissiles - minMissiles) * missiles + 1 - 2 * (ratio - 1) / (maxMissiles - minMissiles))
+end
+
+-- The maximum profit of the base (smallest) ship.
+Calibration.minProfit = Calibration.profit(Vmin)
+
+--Nominal cash needed to trade all stock of a commodity depended on its nominal price.
+Calibration.InvestmentOfPrice = function (price)
+	return Calibration.StockPriceOne / price^(Calibration.PriceExponential - 1)
+end
+
+-- Adding prerequisite feature to ships in order to exclude them from function addRandomShipAdvert of SpaceStation.lua, if player is not their prequisite.
+-- Prerequisite is table for example prerequisites["Sinonatrix"] = {"Amphiesma", "Lunar Shuttle"}
+
+Calibration.prerequisite = function (pship,xship)
+	local prerequisites ={}
+	prerequisites["Lunar Shuttle"] = {"Amphiesma"}
+	prerequisites["Sinonatrix"] = {"Lunar Shuttle"}
+	prerequisites["Mola Mola"] = {"Sinonatrix"}
+	prerequisites["Natrix"] = {"Mola Mola"}
+	prerequisites["Skipjack"] = {"Natrix"}
+	prerequisites["Deneb"] = {"Skipjack"}
+	prerequisites["Bluenose"] = {"Deneb"}
+	prerequisites["AC-33 Dropstar"] = {"Bluenose"}
+	prerequisites["Mola Ramsayi"] = {"AC-33 Dropstar"}
+	prerequisites["Venturestar"] = {"Mola Ramsayi"}
+	prerequisites["Storeria"] = {"Venturestar"}
+	prerequisites["Malabar"] = {"Storeria"}
+	prerequisites["Nerodia"] = {"Malabar"}
+	prerequisites["Vatakara"] = {"Nerodia"}
+	prerequisites["Lodos"] = {"Vatakara"}
+
+	if not prerequisites[xship] then
+		return true
+	else
+		for i, p in pairs(prerequisites[xship]) do
+		if p == pship then
+				return true
+			end
+		end
+		return false
+	end
+end
+
+--------------------- Not used. Just helpfull -------------------------
+
+--Nominal cash needed to trade all stock of a commodity depended on its nominal stock.
+Calibration.InvestmentOfStock = function (stock)
+	return (Calibration.StockPriceOne / stock)^(1 / Calibration.PriceExponential) * stock
+end
+
+--Nominal price of best commodity for a ship with cargo tones free cargo to trade
+Calibration.BestCommodityPrice = function (cargo)
+	return (v2 * Calibration.StockPriceOne / cargo)^(1 / Calibration.PriceExponential)
+end
+
+--Price of 'next' ship depended on current ship's free cargo and value along with 'turns' (or standard units or station visits) needed to acquire it.
+Calibration.ShipPrice = function(currentShipPrice, currentCargo, turns)
+	return Calibration.profit(currentCargo) * turns + currentShipPrice / 2
+end
```

Comment:
I'm confused. If those aren't used, how are they helpful? Is it for console users?

Reply:
**Yes it is for console user to see what can be achieved.** 
Let assume that my proposed exponential is fixed. (_Once decided upon it, it should be fixed. So as all developers can have a steady common reference)_ A ship designer builds a ship with 100t free cargo. He can see by print(require "Calibration".BestCommodityPrice(100)) what is the price of best commodity to be traded by that ship. If he does not like it, he may decide to build a ship with different cargo or even introduce a missing commodity itself for his new ship to trade. (I did it in spreedsheet though, for all ships at once.)

For ShipPrice, ship builder decides how hard will be for the player to buy it. Assume that player's current ship has 70t free cargo, of cource he knows its price (assume 350000c) and want player to be able to buy it in 20 turns or trades if you prefer.  By print(require "Calibration".ShipPrice(350000, 70, 20)) he gets the right price for his ship.